pkg/auth/common/login_handler.go

```diff
@@ -366,21 +430,67 @@ func (h *TanzuLoginHandler) promptAndLoginWithAuthCode(ctx context.Context, auth
 	return wg.Wait
 }
 
+// Returns custom TLS configuration if explicitly provided to the handler,
+// of if persisted cert information associated with the issuer endpoint is found,
+// with the provided information taking precedence over persisted information.
+func (h *TanzuLoginHandler) getTLSConfig() *tls.Config {
+	var savedCertData string
+	var savedSkipVerify bool
+
+	c, _ := config.GetCert(h.issuer)
+
+	if c != nil {
+		savedCertData = c.CACertData
+		savedSkipVerify, _ = strconv.ParseBool(c.SkipCertVerify)
+	}
+
+	if savedSkipVerify || h.tlsSkipVerify {
+		//nolint:gosec // skipTLSVerify: true is only possible if the user has explicitly enabled it
+		return &tls.Config{InsecureSkipVerify: true, MinVersion: tls.VersionTLS12}
+	}
+
+	caCertData := savedCertData
+	if h.caCertData != "" {
+		caCertData = h.caCertData
+	}
+
+	if caCertData != "" {
+		var pool *x509.CertPool
+		var err error
+		pool, err = x509.SystemCertPool()
+		if err != nil || pool == nil {
+			pool = x509.NewCertPool()
+		}
+		pool.AppendCertsFromPEM([]byte(caCertData))
+		return &tls.Config{RootCAs: pool, MinVersion: tls.VersionTLS12}
+	}
+
+	return nil
+}
+
 func (h *TanzuLoginHandler) getTokenUsingAuthCode(ctx context.Context, code string) (*oauth2.Token, error) {
-	// TODO(vuil) support custom CA cert for UAA
 	if h.idpType == config.UAAIdpType {
-		tr := &http.Transport{
-			TLSClientConfig: &tls.Config{InsecureSkipVerify: true}, //nolint:gosec // to update
+		tlsConfig := h.getTLSConfig()
+		if tlsConfig != nil {
+			tr := &http.Transport{
+				TLSClientConfig: tlsConfig,
+			}
```

Comment:
I am not sure if it makes any different or not but I think it would be safe to create the transport from the DefaultTransport and update the TLSClientConfig. For example:

```
transport := http.DefaultTransport.(*http.Transport)
transport.TLSClientConfig = tlsConfig
```

Reply:
Thanks for bringing this up. I will derive a transport from the htp.DefaultTransport.